src/connection/session-manager.js

```diff
@@ -2,13 +2,23 @@ import {
   STORAGE_SESSION_NAME
 } from '../constants'
 
-class Session {
+class SessionManager {
+
+  constructor(type) {
+    this.type = type
+  }
 
   _getStorage() {
-    if(window.localStorage) {
-      return window.localStorage
+    // Chrome can return Access Denied error when trying to access window.localStorage
+    try {
+      switch(this.type) {
+        case 'local': return window.localStorage
+        case 'session': return window.sessionStorage
+        default: return new TestStorage() // used in tests
+      }
+    } catch(e) {
+      return new TestStorage()
```

Comment:
If `Access Denied` happens, they get TestStorage?

Reply:
yes, it basically emulates local storage